app/Livewire/Project/Application/General.php

```
@@ -242,12 +247,31 @@ public function generateDomain(string $serviceName)
-        $this->parsedServiceDomains[$serviceName]['domain'] = $domain;
-        $this->application->docker_compose_domains = json_encode($this->parsedServiceDomains);
+        $sanitizedKey = str($serviceName)->slug('_')->toString();
+        $this->parsedServiceDomains[$sanitizedKey]['domain'] = $domain;
+
+        // Convert back to original service names for storage
+        $originalDomains = [];
+        foreach ($this->parsedServiceDomains as $key => $value) {
+            // Find the original service name by checking parsed services
+            $originalServiceName = $key;
+            if (isset($this->parsedServices['services'])) {
+                foreach ($this->parsedServices['services'] as $originalName => $service) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$service'. (Unused Code Rules)

(UnusedLocalVariable)